src/RadixTree.elm

```diff
@@ -1,107 +1,67 @@
 module RadixTree exposing
-    ( empty, singleton, insert, insertWith, insertUnOrdered
-    , orderedIntersect, orderedRemove, unOrderedIntersect, unOrderedRemove
+    ( RadixTree, toTree
+    , empty, singleton, insert
     )
 
-{-| Radix tree as defined is defined in Wikipedia: <https://en.wikipedia.org/wiki/Radix_tree>
+{-| Build an ordered Radix tree.
 
-A data structure that represents a space-optimized trie (prefix tree) in which
-each node that is the only child is merged with its parent
 
-It would be easier to follow by example, or try it on a helpful app like
-this one: <https://www.cs.usfca.edu/~galles/visualization/RadixTree.html>
+# Structure
 
-Say we have the following list of chars:
-
-    [ [ "r", "o", "m", "a", "n", "e" ]
-    , [ "r", "o", "m", "u", "l", "u", "s" ]
-    , [ "r", "u", "b", "e", "n", "s" ]
-    , [ "r", "u", "b", "e", "r" ]
-    , [ "r", "u", "b", "i", "c", "o", "n" ]
-    , [ "r", "u", "b", "i", "c", "u", "n", "d", "u", "s" ]
-    ]
-
-We would like to represent this as a Radix tree. Where letters that repeat are
-in parent nodes, and the rest are in child nodes.
-
-It would look like this:
-
-  - r
-      - ub
-          - ic
-              - undus
-              - on
-          - e
-              - r
-              - ns
-      - om
-          - ulus
-          - ane
-
-This package allows inserting values to the tree in an Ordered and Un-ordered manner.
-The ordered one is presented above. The "order" is in relation to the position of each
-letter. That is, when we look for the letter "r" that is the beginning of the list
-, on an Ordered list, we match it only to other "r" letters that are also at the beginning.
-
-An Un-ordered insert on the other hand doesn't care about the position of the chars, but rather if they exist or not.
-
-Here's an example of how an un-ordered insert would result, assuming we have this list
-of Int.
-
-    [ [ 1, 2, 3 ]
-    , [ 2, 1, 4 ]
-    , [ 4, 5, 6 ]
-    ]
-
-This will result with:
-
-  - 1, 2
-      - 3
-      - 4
-          - 5
-          - 6
+@docs RadixTree, toTree
 
 
 # Build
 
-@docs empty, singleton, insert, insertWith, insertUnOrdered
+@docs empty, singleton, insert
 
+-}
 
-# Lists manipulation
+import Internal.Utils exposing (insertWith, orderedIntersect, orderedRemove)
+import Tree
 
-@docs orderedIntersect, orderedRemove, unOrderedIntersect, unOrderedRemove
 
+{-| Represents an ordered Radix tree. Under the hood we use [Tree](https://package.elm-lang.org/packages/zwilias/elm-rosetree/latest/Tree)
 -}
+type RadixTree a
+    = RadixTree (Tree.Tree (List a))
 
-import List.Extra
-import Tree
-import Tree.Zipper as TreeZipper
+
+{-| Extract the `Tree` from the `RadixTree`.
```

Comment:
This is where I would mention `[Tree](https://package.elm-lang.org/packages/zwilias/elm-rosetree/latest/Tree)`, not in the documentation of the `RadixTree` type. Maybe you can also explain why someone would want this (for querying, traversing, ...)